change-validation/src/test/scala/com/normation/plugins/changevalidation/api/ChangeRequestApiTest.scala

```diff
@@ -0,0 +1,659 @@
+      )
+    )
+  }
+
+  val mockDirectives  = restTestSetUp.mockDirectives
+  val mockNodeGroups  = restTestSetUp.mockNodeGroups
+  val mockRules       = restTestSetUp.mockRules
+  val mockGlobalParam = new MockGlobalParam()
+
+  val mockServices = new MockServices(
+    Map(
+      Validation.id -> List(
+        ConfigurationChangeRequest(
+          ChangeRequestId(1),
+          None,
+          ChangeRequestInfo("first cr directive", "My directive first change"),
+          Map(
+            DirectiveId(DirectiveUid("directive_001")) -> DirectiveChanges(
+              DirectiveChange(
+                None,
+                DirectiveChangeItem(
+                  actor,
+                  DateTime.parse("2023-01-01T00:00:00.000Z"),
+                  Some("directive_001 change reason"),
+                  AddDirectiveDiff(TechniqueName("packageManagement"), mockDirectives.directives.pkgDirective)
+                ),
+                List.empty
+              ),
+              List.empty
+            )
+          ),
+          Map.empty,
+          Map.empty,
+          Map.empty
+        ),
+        ConfigurationChangeRequest(
+          ChangeRequestId(2),
+          None,
+          ChangeRequestInfo("second cr directive", "My directive second change"),
+          Map(
+            DirectiveId(DirectiveUid("directive_001")) -> DirectiveChanges(
+              DirectiveChange(
+                Some(
+                  (
+                    TechniqueName("packageManagement"),
+                    mockDirectives.directives.pkgDirective,
+                    Some(directiveSectionSpec)
+                  )
+                ),
+                DirectiveChangeItem(
+                  actor,
+                  DateTime.parse("2023-02-02T00:00:00.000Z"),
+                  Some("directive_001 change reason"),
+                  ModifyToDirectiveDiff(
+                    TechniqueName("packageManagement"),
+                    mockDirectives.directives.pkgDirective,
+                    Some(directiveSectionSpec)
+                  )
+                ),
+                List(
+                  DirectiveChangeItem(
+                    actor,
+                    DateTime.parse("2023-02-02T00:00:00.000Z"),
+                    Some("directive_001 another change reason"),
+                    ModifyToDirectiveDiff(
+                      TechniqueName("packageManagement"),
+                      mockDirectives.directives.pkgDirective.copy(
+                        name = "pkg_directive_001",
+                        shortDescription = "testing directive change",
+                        longDescription = "testing directive change",
+                        priority = 1,
+                        isSystem = true,
+                        parameters = mockDirectives.directives.pkgDirective.parameters ++ Map(("PACKAGE_LIST", Seq("curl")))
+                        // ignored changes :
+                        // policyMode = Some(PolicyMode.Audit),
```

Comment:
policy mode should not be ignored. Why is it so ? 
(even tag, I'm not sure) (they should, they are in eventlog: https://issues.rudder.io/issues/15019)

Reply:
Both `policyMode` and `tags` changes would be taken into account : https://github.com/Normation/rudder/blob/master/webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/domain/policies/DirectiveDiff.scala#L86, but in the API response (yaml tests), the diff for both of them were never displayed : https://github.com/Normation/rudder/blob/master/webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/apidata/RestDataSerializer.scala#L535

It could be added without risk, but I didn't because it would mean I'd also have to add these new fields to the 'json datatype' and also change the json tests... Is it okay to keep those lines as comments or should I make the change ?

Reply:
I think I would be better off in another PR, because I tried to implement it and the change needs a consequent adaptation of the current code (adapt the `{"from":...,"to":...}` json serialisation because it does not display 'null', and add corresponding lines of code/tests)